Source/Dafny/Compilers/SinglePassCompiler.cs

```diff
@@ -4863,8 +4881,12 @@ private class GetterSetterLvalueImpl : ILvalue {
         var e = (LambdaExpr)expr;
 
         wr = CaptureFreeVariables(e, false, out var su, inLetExprBody, wr, wStmts);
+        if (su != Substituter.EMPTY && TargetLambdasRestrictedToExpressions) {
+          wStmts = wr.Fork();
+          wr = EmitReturnExpr(wr);
+        }
```

Comment:
Just noting for the record that as I'm better appreciating our solution so far for `TargetLambdasRestrictedToExpressions`, I definitely see more speed bumps down the road for edge cases we don't handle correctly, because we're not doing this "post hoc clean up" for every possible method call that might need it. I won't likely see an immediate alternative and don't want to block this proven progress, but I do want to revisit before we go too much further.

Reply:
Yes, those changes are unlikely to be comprehensive, but I was hesitant to introduce additional untested changes.